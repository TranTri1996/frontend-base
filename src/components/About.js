import React from 'react';

const About = () => {
    return (
        <div><h3>This is About component</h3></div>
    );
};

export default About;
