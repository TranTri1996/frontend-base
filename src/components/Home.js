import React, { useState, useCallback, useMemo } from 'react';
import TestEvaluationHook from './TestEvaluationHook/TestEvaluationHookComponent';
import './Home.scss';

const Home = () => {
  const [count, setCount] = useState(0);
  const handleOnClick = useCallback(() => setCount(count + 1));

  return (
    <>
      <h2>this is Home component</h2>
      <button onClick={handleOnClick}>Click</button>
      <TestEvaluationHook count={count} />
    </>
  );
};

export default Home;
