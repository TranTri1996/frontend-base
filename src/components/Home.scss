.post{
  background-color: #eee;
  margin: 2rem;
  padding: 1rem;
}
.pagination{
  margin: 1rem auto;
  list-style: none;
  display: flex;
  justify-content: space-evenly;
  width: 50%;
}
.active{
  border: 1px solid black;
  border-radius: 100%;
  padding: 0 3px;
  outline: none;
}