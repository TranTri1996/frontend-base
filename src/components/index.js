export * from './SideBar';
export * from './ToolBar';
export * from './Home';
export * from './App';
export * from './NotFound';