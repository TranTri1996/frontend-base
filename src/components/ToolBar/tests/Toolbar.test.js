import React from 'react';
import { expect } from '@jest/globals';
import renderer from 'react-test-renderer';
import { ToolBar } from 'components';

describe('Toolbar component', () => {
  it('should render successful', () => {
    const toolbarSnapshot =   renderer.create(<ToolBar />      ).toJSON();
    expect(toolbarSnapshot).toMatchSnapshot();
  });
});