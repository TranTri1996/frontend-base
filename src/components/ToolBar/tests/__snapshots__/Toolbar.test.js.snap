// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component should render successful 1`] = `
<div
  className="toolbar"
>
  This is ToolBar
</div>
`;
